### PR TITLE
Kirjoitin yleisluontoisen postauksen neuroverkoista

### PR DESCRIPTION
**Pull request by jvuorinen**
...ja lisäsin configiin omat tiedot. Tekstiä on ds-tiimissä vertaisarvioitu.

### CODE REVIEW

**Comment by jvuorinen** [comment]
Julkaisupvm vaihdettu

File: _posts/2018-02-12-deep-learning-popularity.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+author: jessevuorinen
+excerpt: Thoughts on why deep learning models are popular, and some tips on how to get started with them.
+tags:
+- data science
+- neural network
+- supervised learning
+- artificial intelligence
+---
+
+Deep learning (i.e. big neural networks) is at the forefront of the ongoing trend of artificial intelligence and data science. Last year, a neural network based AI beat a human grandmaster in Go, one of the hardest board games. From machine translators to self-driving vehicles, it seems deep neural networks can be found in most state-of-the-art AI solutions. In our client projects, we have leveraged deep learning in image recognition problems and multivariate time series forecasting, for example. What makes it possible for neural networks to be able to generalize to such a broad range of problems? Why are they so useful?
```
Comment (eerohele) [inline _posts/2018-02-12-deep-learning-popularity.md]: “What makes it possible…” is OK, but something like "Why are neural networks applicable to such a broad range of problems?" might be more succinct.

File: _posts/2018-02-12-deep-learning-popularity.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+From a distance, a neural network (and most other supervised machine learning algorithms) can be seen as a device that takes in numerical inputs and spits out numerical outputs. When the model is built, the general structure of what is inside the device, as well as the structure of its inputs and outputs, is specified. At this point, the model already "works" in the sense that it can take inputs to produce outputs, but the results are random.
```
Comment (eerohele) [inline _posts/2018-02-12-deep-learning-popularity.md]: To me, “from a distance” implies physical distance — maybe “On a high level”? Not a necessary change, though, I don't think.

File: _posts/2018-02-12-deep-learning-popularity.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+From a distance, a neural network (and most other supervised machine learning algorithms) can be seen as a device that takes in numerical inputs and spits out numerical outputs. When the model is built, the general structure of what is inside the device, as well as the structure of its inputs and outputs, is specified. At this point, the model already "works" in the sense that it can take inputs to produce outputs, but the results are random. 
+
+Then, the model is trained by continuously feeding it with actual data, that is, correct answers to the problem at hand. During this training process the parameters of the model (the bolts and cogs inside the device) are adjusted in very small increments. In time, the algorithm converges and learns the relationships in the training data. In the end, the model learns how to map input data to outputs using a similar logic that underlies the training data it was fed. After training, the model can be used to make predictions based on new input data, something that the model has not seen before.
+
+### What goes in, what goes out?
```
Comment (eerohele) [inline _posts/2018-02-12-deep-learning-popularity.md]: I think second-level titles usually use two hash marks (##).

File: _posts/2018-02-12-deep-learning-popularity.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Second, since neural networks comprise of chained little functions that perform nonlinear transformations at each step, they are inherently nonlinear models. This allows them to model many problems better since many real-world relationships are nonlinear. (For example, the area of a square shaped field increases exponentially instead of linearly as its width increases.)
```
Comment (eerohele) [inline _posts/2018-02-12-deep-learning-popularity.md]: If we want to get really nit-picky: [~comprise of~](https://www.npr.org/2015/03/12/392568604/dont-you-dare-use-comprised-of-on-wikipedia-one-editor-will-take-it-out) either “comprise” (without “of”) or “consist of”.

File: _posts/2018-02-12-deep-learning-popularity.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+Third, the vanilla version of a neural network that has been discussed so far, can be adjusted to make it a better fit to certain problems. Convolutional neural networks, for example, are good at making broad abstractions from very detailed inputs and work especially well for image recognition problems. In recurrent neural networks, the neuron layers have feedback loops, which essentially means that the network is able to remember previous inputs. This trait makes them a good fit for time series forecasting and natural language processing.
```
Comment (eerohele) [inline _posts/2018-02-12-deep-learning-popularity.md]: “so far, can” -> “so far can”

**Pull request closed without merging**